march_gazebo_plugins/src/com_controller_plugin.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019 Project March.
+
+#include <com_controller_plugin.h>
+#include <typeinfo>
+
+namespace gazebo
+{
+void ComControllerPlugin::Load(physics::ModelPtr _parent, sdf::ElementPtr /*_sdf*/)
+{
+  // Make sure the ROS node for Gazebo has already been initialized
+  if (!ros::isInitialized())
+  {
+    ROS_FATAL_STREAM("A ROS node for Gazebo has not been initialized, unable to load "
+                     "plugin. "
+                     << "Load the Gazebo system plugin 'libgazebo_ros_api_plugin.so' in "
+                        "the gazebo_ros package)");
+    return;
+  }
+
+  // Initialise variables
+  this->model_ = _parent;
+  this->controller_.reset(new WalkController(this->model_));
+
+  this->subgait_name_ = "home_stand";
+  this->stable_side_ = "left";
+
+  // Listen to the update event. This event is broadcast every
+  // simulation iteration.
+  this->update_connection_ = event::Events::ConnectWorldUpdateBegin(std::bind(&ComControllerPlugin::onUpdate, this));
+
+  // Create our ROS node.
+  this->ros_node_ = std::make_unique<ros::NodeHandle>(ros::NodeHandle("gazebo_client"));
```

Comment:
`std::make_unique` forwards its given arguments to the constructor of the given type. So, now you are creating a `ros::NodeHandle` and passing that to the constructor, which will copy the given handle. For simplicity, and less copies, you can just give the node handle name.

Furthermore, shouldn't this name be more descriptive, like **com_controller_plugin**?
```suggestion
  this->ros_node_ = std::make_unique<ros::NodeHandle>("com_controller_plugin");
```